eth/backend.go

```diff
@@ -483,6 +483,17 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 						logItems = append(logItems, eth.ProtocolToString[protocol], strconv.Itoa(count))
 					}
 					logger.Info("[p2p] GoodPeers", logItems...)
+
+					if err := chainKv.Update(context.Background(), func(tx kv.RwTx) error {
+						if err = stagedsync.UpdateMetrics(tx); err != nil {
```

Comment:
I searched for `syncMetrics` variable use and found that it get updated inside `eth/stagedsync/stage.go:StageState.Update` - and this is only way to update stage progress in DB. 
So, it's a bit unclear for me: if we updating metrics there - why need update theme also here. 
Also `StageState.Update` setting value of metric even if RwTx was not committed, while external RoTx (which you open here) can see only committed changes. 
On non-chain-tip we do rare commits, on chain-tip - we commit every block (after stage_finish).

Reply:
If you look at `eth/stagedsync/stage.go:StageState.Update` you'll see that it updates metrics and save to DB calling `return stages.SaveStageProgress(db, s.ID, newBlockNum)`, it calls when stage is finished. But consensus layer skip bodies stage and save data to DB from`eth/stagedsync/stage_snapshots.go:FillDBFromSnapshots` it calls `stages.SaveStageProgress(tx, stage, blocksAvailable)`. So the good way to update metrics is to put it inside `stages.SaveStageProgress` but it's require refactoring E3 code and I don't know how to test it properly, that is why I put update there. It is possible to put metrics update inside `eth/stagedsync/stage_snapshots.go:FillDBFromSnapshots` but it is one of many places where `stages.SaveStageProgress` is called.

Reply:
Changed as discussed